File: backend/src/routes/assess.ts
import { Router, Request, Response } from 'express';
import { v4 as uuid } from 'uuid';
import type { ProjectInput } from '../types';
import { runAssessmentPipeline } from '../services/assessmentPipeline';

const router = Router();

router.post('/', async (req: Request, res: Response) => {
  const input: ProjectInput = req.body;

  if (!input.address || !input.roofAreaM2) {
    return res.status(400).json({ error: 'address and roofAreaM2 are required' });
  }

  const projectId = uuid();

  try {
    const result = await runAssessmentPipeline(projectId, input);
    return res.json(result);
  } catch (err) {
    console.error('Assessment pipeline error:', err);
    return res.status(500).json({ error: 'Assessment failed' });
  }
});

router.get('/:id', async (req: Request, res: Response) => {
  // In production: look up from DB. For MVP, return 404.
  return res.status(404).json({ error: 'Assessment not found (persistence not yet implemented)' });
});

export default router;

File: backend/app/routers/assess.py
"""POST /api/assess — port of routes/assess.ts."""
from __future__ import annotations

from uuid import uuid4

from fastapi import APIRouter, HTTPException

from app.models import AssessmentResult, ProjectInput
from app.services.assessment_pipeline import run_assessment_pipeline

router = APIRouter()


@router.post("", response_model=AssessmentResult)
async def assess(inp: ProjectInput) -> AssessmentResult:
    if not inp.address or not inp.roofAreaM2:
        raise HTTPException(status_code=400, detail="address and roofAreaM2 are required")

    project_id = str(uuid4())
    try:
        return await run_assessment_pipeline(project_id, inp)
    except Exception as exc:
        import logging
        logging.getLogger(__name__).error("Assessment pipeline error: %s", exc)
        raise HTTPException(status_code=500, detail="Assessment failed")


@router.get("/{assessment_id}", response_model=dict)
async def get_assessment(assessment_id: str) -> dict:
    # In production: look up from DB. For MVP, return 404.
    raise HTTPException(
        status_code=404,
        detail="Assessment not found (persistence not yet implemented)",
    )
